frontend/src/pages/Login.js

```diff
@@ -0,0 +1,172 @@
+import React, { useEffect, useState } from "react";
+import Avatar from '@material-ui/core/Avatar';
+import Button from '@material-ui/core/Button';
+import CssBaseline from '@material-ui/core/CssBaseline';
+import TextField from '@material-ui/core/TextField';
+import Link from '@material-ui/core/Link';
+import Paper from '@material-ui/core/Paper';
+import Grid from '@material-ui/core/Grid';
+import MuiAlert from '@material-ui/lab/Alert';
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
+import Typography from '@material-ui/core/Typography';
+import { makeStyles } from '@material-ui/core/styles';
+import { Configuration, PublicApi } from '@ory/kratos-client';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    height: '100vh',
+  },
+  image: {
+    backgroundImage: 'url(https://source.unsplash.com/random)',
+    backgroundRepeat: 'no-repeat',
+    backgroundColor:
+      theme.palette.type === 'light' ? theme.palette.grey[50] : theme.palette.grey[900],
+    backgroundSize: 'cover',
+    backgroundPosition: 'center',
+  },
+  paper: {
+    margin: theme.spacing(8, 4),
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+  },
+  avatar: {
+    margin: theme.spacing(1),
+    backgroundColor: theme.palette.secondary.main,
+  },
+  csrf: {
+    visibility: 'hidden'
+  },
+  form: {
+    width: '100%', // Fix IE 11 issue.
+    marginTop: theme.spacing(1),
+  },
+  submit: {
+    margin: theme.spacing(3, 0, 5),
+  },
+}));
+
+function Alert(props) {
+  return <MuiAlert elevation={6} variant="filled" {...props} />;
+}
+
+function Login() {
+  const classes = useStyles();
+  
+  const [formUsernamePasswordAction, setFormUsernamePasswordAction] = useState();
+  const [csrfUsernamePasswordToken, setCsrfUsernamePasswordToken] = useState();
+  const [loginMessages, setLoginMessages] = useState(undefined);
+
+  const kratos = new PublicApi(new Configuration({ basePath: 'http://127.0.0.1:4433' }));
+  
+  useEffect(() => {
+    if (!(new URL(document.location)).searchParams.get("flow")) {
+        window.location.href = "http://127.0.0.1:4433/self-service/login/browser";
+    }
+    const flowId = (new URL(document.location)).searchParams.get("flow");
+    kratos.getSelfServiceLoginFlow(flowId)
+        .then(({ status, data: flow }) => {
+            if (status === 404 || status === 410 || status === 403) {
```

Comment:
```suggestion
            if ([401, 403, 404].includes(status)) {
```
Much less verbose :) And I guess it's 401 vs 410, right?